velox/exec/TopNRowNumber.cpp

```diff
@@ -530,37 +555,53 @@ bool TopNRowNumber::isNewPartition(
   return false;
 }
 
-void TopNRowNumber::setupNextOutput(
+// Compares the partition keys for new partitions.
+bool TopNRowNumber::isNewPartition(
```

Comment:
Let's merge isNewPartition and compareSpillRowColumns? And call this isNewPartitionFromSpill?

Reply:
Same response as first question. We need another function for comparing order_by columns for rank and so compareSpillRowColumns is abstracted.